### PR TITLE
MGMT-8370 - Replace the client on the service with the new generated one

### PR DESCRIPTION
**Pull request by eliorerz**
# Assisted Pull Request

## Description
Replace the python client  artifact on the service image with the artifact we upload to PYPI (versionised)
- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @osherdp

### CODE REVIEW

File: tools/client_package_initializer.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def _execute(cls, data: str) -> Dict[str, Dict]:
         exec(data, d)
         return d
 
-    def dump(self) -> "SetupInitializer":
+    def dump(self, build=False) -> "SetupInitializer":
```
Comment (osherdp) [inline tools/client_package_initializer.py:125]: where do you make use of ``build``?

File: hack/generate.sh [hunk 2]
```diff
@@ -33,8 +33,9 @@ function generate_python_client() {
         -v "${__root}"/tools/generate_python_client.sh:/script.sh:ro,Z \
         -e SWAGGER_FILE=/swagger.yaml -e OUTPUT=/local/assisted-service-client/ \
         quay.io/ocpmetal/swagger-codegen-cli:2.4.15 /script.sh
-     cd "${dest}"/assisted-service-client/ && python3 setup.py sdist --dist-dir "${dest}"
      cd "${dest}"/assisted-service-client/ && python3 "${__root}"/tools/client_package_initializer.py "${dest}"/assisted-service-client/  https://github.com/openshift/assisted-service --build
+     rm -f "${dest}"/assisted-service-client-*.tar.gz
```
Comment (osherdp) [inline hack/generate.sh:37]: I don't understand
Why do we create it in the first place if we're not using it?

File: tools/client_package_initializer.py [hunk 3]
```diff
@@ -122,7 +122,7 @@ def _execute(cls, data: str) -> Dict[str, Dict]:
...
         """
         Dump setup.py into file
```
Comment (osherdp) [inline tools/client_package_initializer.py:127]: update, as it's doing something else now

File: tools/client_package_initializer.py [hunk 4]
```diff
@@ -122,7 +122,7 @@ def _execute(cls, data: str) -> Dict[str, Dict]:
...
         :return:
```
Comment (osherdp) [inline tools/client_package_initializer.py:128]: delete this line

File: tools/client_package_initializer.py [hunk 5]
```diff
@@ -132,15 +132,35 @@ def dump(self) -> "SetupInitializer":
...
         with open(self._setup_path, "w") as f:
             f.write(Setup.FORMAT.format(**self._setup_data))
+
+        self.build("bdist_wheel")
+
+        with open(self._setup_path, "w") as f:
+            f.write(Setup.FORMAT.format(**self._setup_data))
+
+        cmd = f"python3 {self._setup_path} check"
+        subprocess.check_output(cmd.split())
```
Comment (osherdp) [inline tools/client_package_initializer.py:142]:
```suggestion
        subprocess.check_output(["python3", self._setup_path, "check"])
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: tools/client_package_initializer.py (hunk 4 above)
Comment (eliorerz) [reply tools/client_package_initializer.py]: Done

File: hack/generate.sh (hunk 2 above)
Comment (eliorerz) [reply hack/generate.sh]: We don't need that anymore. Removed that line.

File: tools/client_package_initializer.py (hunk 1 above)
Comment (eliorerz) [reply tools/client_package_initializer.py]: Removed it

**Review by osherdp — APPROVED** [review]

**Pull request merged**